[SYCLomatic] Update README.md with test suite and description. (#351)
oneapi-src/SYCLomatic-test: README.md

| name         | Name of test suite, also **${dir}/${name}.xml** is configuration file for this suite.           |
| opts         | Specifies the test option list for each test suite. Test options can be: **option_cpu**, **option_gpu**, **option_usmnone_cpu**, **option_cuda** and so on. Each test option defines the target device to run and extra c2s options during migration, more details in the [Define option for a test suite](#define-option-for-a-test-suite-option_mappingjson). |

### Current test suite in SYCLomatic-test
| Test suite name | Description| Link with the CI|
| ------------ | ------------------------------------------------------------ |---|

> |api_coverage | The APIs migrated by SYCLomatic currently include only the relevant APIs for cublas migration. Please visit https://github.com/oneapi-src/SYCLomatic/tree/SYCLomatic/clang/test/dpct to access more migration test cases. |test-api_coverage|

Reviewer: ```suggestion
|api_coverage | The test suite includes API migration tests cases.  Also reference to https://github.com/oneapi-src/SYCLomatic/tree/SYCLomatic/clang/test/dpct for more API migration test cases. |test-api_coverage|
```

> |behavior_tests | The test suite includes both positive and negative behavior tests for the SYCLomatic tool, target to validate tool's expected behavior. |test-behavior_tests|

Reviewer: ### 
```suggestion
|behavior_tests | The test suite includes both positive and negative behavior tests, target to validate the expected behavior of the SYCLomatic tool. |test-behavior_tests|
```

> |behavior_tests | The test suite for the SYCLomatic tool includes both positive and negative behavior tests to validate its expected behavior. |test-behavior_tests|

Reviewer: ```suggestion
|behavior_tests | The test suite includes both positive and negative behavior tests for the SYCLomatic tool, target to validate tool's expected behavior. |test-behavior_tests|
```

> |help_function | This test suite encompasses all the help functions developed by SYCLomatic that facilitate smoother code migration. If you have enabled any new help functions, please ensure that corresponding test cases are added to this suite. | test-help_function|

Reviewer: Please update the test suite name from:  from help_function ==> helper_function_tests
help function(s) ==> helper function(s)
add a link to the helper function files.

> |features | The existing suite covers all implementation features in SYCLomatic. If you introduce a new feature, please ensure to include corresponding end-to-end test cases in this suite. This will ensure comprehensive coverage of the new feature within SYCLomatic. |test-features|

Reviewer: Please add examples for what is a new feature.

> |features | The current test suite already covers all the implementation features in SYCLomatic. However, if you are introducing a new feature, such as enabling new APIs or adding additional tool functionalities, please make sure to include the corresponding end-to-end test cases in the suite. By doing so, we can ensure comprehensive coverage of the new feature within SYCLomatic. |test-features|

Reviewer: ```suggestion
|features | The current test suite contains test cases for features of SYCLomatic. If a new feature is implemented, like enabling migration for new API(s), adding additional tool functionality, please make sure to include the corresponding end-to-end test cases in the suite. |test-features|
```

> |regressions|The suite includes regression test cases that capture any issues encountered during the development process.|test-regressions|

Reviewer: ```suggestion
|regressions|The suite includes regression test cases which are captured during regular test. |test-regressions|
```

|samples|The suite consists of small end-to-end samples that cover various functionalities and scenarios.|test-samples|

> |user_guide_samples|The suite includes test cases derived from the SYCLomatic user guide, ensuring that the functionality described in the guide is thoroughly tested.|test-user_guide_samples|

Reviewer: ```suggestion
|user_guide_samples|The suite includes test cases derived from the SYCLomatic user guide, ensuring that the functionality described in the guide is thoroughly tested.|test-user_guide_samples|
```
```suggestion
|user_guide_samples|The suite includes test cases derived from the SYCLomatic user guide. |test-user_guide_samples|
```
